File: programming_languages/python/data_structures_implementation/stack/linear_stack.py
class stack( object ):
	def __init__(self):
		self.top = -1
		self.size = int(input("Enter size of stack:\t"))
		self.arr = [None]*self.size

	def push(self):
		if self.top < self.size-1:
			val = int(input("Enter value to be inserted\t"))
			self.top+=1
			self.arr[self.top] = val
			print(self.arr[self.top] , "INSERTED")
		else:
			print("OVERFLOW")

	def pop(self):
		if self.top > -1:
			print( self.arr[self.top] , "deleted")
			self.top-=1
		else:
			print("UNDERFLOW")

	def display(self):
		if self.top > -1:
			high = self.top
			low = -1
			for x in range(high,low,-1):
				print(self.arr[x])
		else:
			print ("EMPTY")

def controller():
	user_stack = stack()
	choice = None
	again = True

	while( again ):
		print("1). PUSH")
		print("2). POP")
		print("3). DISPLAY")
		print("4). EXIT")
		choice = int(input("ENTER CHOICE\t"))

		if choice == 1:
			user_stack.push()
		elif choice == 2:
			user_stack.pop()
		elif choice == 3:
			user_stack.display()
		elif choice == 4:
			print("GOODBYE")
			again = None
		else:
			print("ERROR CHOICE")

controller()

File: programming_languages/cpp/data_structures_implementation/stacks/linearStack.cpp
#include <iostream>

using namespace std;

class stack{

	int top, size, value, *a;

	public:
		stack( ){
			top = -1;
			cout << "ENTER SIZE\t";
			cin >> size;

			a = new int [size];
		}
		void push( ){
			if( top < size - 1  ){
				cout << "ENTER NEW ELEMENT\t";
				cin >> value;

				a[++top] = value;

				cout << "INSERTED" << endl;
			}else{
				cout << "OVERFLOW" << endl ;
			}
		}
		void pop( ){
			if( top > -1 ){
				cout << a[top] << "\tDELETED" << endl;
				--top;
			}else{
				cout << "UNDERFLOW" << endl ;
			}
		}
		void display( ){

			if( top > -1){
				for (int i = top; i > -1; --i)
					cout << "stack[" << i+1 << "] = " << a[i] << endl;
			}
			else{
				cout << "EMPTY" << endl;
			}

		}
};


int main() {

	stack st;

	int choice;
	bool again = true;

	while( again ){
		
		cout << "1). PUSH" << endl;
		cout << "2). POP" << endl;
		cout << "3). DISPLAY" << endl;
		cout << "4). EXIT" << endl;
		cout << "ENTER CHOICE\t";
		cin >> choice;

		switch( choice ){
			case 1:
				st.push();
				break;
			case 2:
				st.pop();
				break;
			case 3:
				st.display();
				break;
			case 4:
				again = false;
				break;
			default:
				cout << "ERROR CHOICE" << endl;
		}
	}

	return 0;
}
